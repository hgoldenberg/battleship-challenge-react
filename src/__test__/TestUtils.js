/* eslint-disable react/prop-types */
/* eslint-disable max-len */
import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render as rlRender } from '@testing-library/react';
import aplicationReducer from '../redux/reducers/index';

const store = createStore(aplicationReducer);

export const ProviderMock = ({ children }) => (
  <Provider store={store}>
    {children}
  </Provider>
);

const Wrapper = ({ children }) => <ProviderMock>{children}</ProviderMock>;

export const renderWithRedux = (component, renderOptions) => rlRender(component, { wrapper: Wrapper, ...renderOptions });