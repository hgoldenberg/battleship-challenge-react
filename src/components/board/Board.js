import React, { useState } from 'react';
import PropTypes from 'prop-types';
import helpers from '../../utils/Helpers';
import { CELL_ID_VALUE } from '../../utils/Constants';
import { Container, CellDiv, Row } from './styles';

const Cell = (props) => {
    const getCellColor = () => {
        if (props.mark) {
            return '#9C9B9A';
        }
        switch(props.id) {
            case CELL_ID_VALUE.DEFAULT:
                return '#D3D3D3';
            case CELL_ID_VALUE.WATER:
                return '#0077B3';
            case CELL_ID_VALUE.HIT:
                return '#FF9933';
            case CELL_ID_VALUE.DESTROYED:
                return 'red';
            default:
                if (!props.showShip) {
                    return 'grey';
                } return '#D3D3D3'
        }
    }

    const handleClick = () => {
        if (!(props.id == CELL_ID_VALUE.WATER || props.id == CELL_ID_VALUE.HIT || props.id == CELL_ID_VALUE.DESTROYED)) {
          props.onClick();
        }
      };

      return (
        <CellDiv color={getCellColor()} onMouseOver={() => props.onMouseHover()} onClick={() => handleClick()} />
      );
    };

const Board = (props) => {
    const [positionsToMark, setPositionsToMark] = useState(undefined);

    const markPositions = (row, col) => {
        setPositionsToMark(undefined);
        if (props.shipSelected) {
          const positions = helpers.getShipPositions(props.board, props.shipSelected.size, row, col, props.shipOrientation);
          if (positions) {
            setPositionsToMark(positions);
          } else {
                setPositionsToMark(undefined);
            }
        } 
    }

    const getBoard = () => {
        let mark;
        const board = props.board.map((row, x) => (
          <Row key={x}>
            {row.map((column, y) => {
              mark = positionsToMark && positionsToMark.findIndex((pos) => pos.row == x && pos.col == y) !== -1;
              return (
                <div key={y}>
                  <Cell
                    showShip={props.cpu}
                    code={props.board[x][y].code}
                    id={props.board[x][y].id}
                    onMouseHover={() => markPositions(x, y)}
                    mark={mark}
                    onClick={() => { props.click && props.onClickBoard({ row: x, col: y }); }}
                  />
                </div>
              );
            })}
          </Row>
        ));
        return board;
      };

    return(
        <Container>
            {getBoard()}
        </Container>
    )
}

Board.propTypes = {
    board: PropTypes.array,
};

export default Board; 