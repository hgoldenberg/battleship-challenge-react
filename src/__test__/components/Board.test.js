/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { renderWithRedux } from '../TestUtils';
import Board from '../../components/board/Board';

const props = {
  board: [],
  cpu: false,
  click: true,
  onClickBoard: () => {},
};

describe('<Board/>', () => {
  test('board component render', () => {
    renderWithRedux(<Board {...props} />);
  });
});

describe('Board snapshot', () => {
  test('Check the Board Snapshot', () => {
    const { container } = renderWithRedux(<Board {...props} />);
    expect(container).toMatchSnapshot();
  });
});