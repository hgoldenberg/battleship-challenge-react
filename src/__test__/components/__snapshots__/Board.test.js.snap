// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board snapshot Check the Board Snapshot 1`] = `
<div>
  <div
    class="sc-AxjAm hZacsw"
  />
</div>
`;
